nat/service_iptables.go

```diff
@@ -18,111 +18,172 @@
 package nat
 
 import (
-	"fmt"
-	"strings"
+	"strconv"
 	"sync"
 
+	"github.com/mysteriumnetwork/node/firewall/iptables"
+	"github.com/mysteriumnetwork/node/utils"
 	"github.com/mysteriumnetwork/node/utils/cmdutil"
 	"github.com/pkg/errors"
 	"github.com/rs/zerolog/log"
-
-	"github.com/mysteriumnetwork/node/config"
 )
 
 type serviceIPTables struct {
 	mu        sync.Mutex
-	rules     map[RuleForwarding]struct{}
+	rules     []iptables.Rule
 	ipForward serviceIPForward
 }
 
-func (service *serviceIPTables) Add(rule RuleForwarding) error {
-	service.mu.Lock()
-	defer service.mu.Unlock()
+const (
+	chainForward     = "FORWARD"
+	chainPreRouting  = "PREROUTING"
+	chainPostRouting = "POSTROUTING"
+)
 
-	if _, ok := service.rules[rule]; ok {
-		return errors.New("rule already exists")
-	}
-	service.rules[rule] = struct{}{}
+// Setup sets NAT/Firewall rules for the given NATOptions.
+func (svc *serviceIPTables) Setup(opts Options) (appliedRules []interface{}, err error) {
```

Comment:
Add unit test

Reply:
Unit test would require to at least run actual iptables and see if the packets are routed/redirected properly. It would make more sense to actually do an end to end test case for this instead.